testing/mocks.go

```diff
@@ -163,3 +163,31 @@ func (g *Getter) Get(url string) ([]byte, error) {
 	}
 	return v, nil
 }
+
+// VariableResponseGetter is a mock for HTTPSGetter interface that sequentially
+// returns the configured responses, independent of the provided URL.
+type VariableResponseGetter struct {
+	// callCount is used to return the respective responses
+	callCount     int
+	ResponseBody  [][]byte
+	ResponseError []error
+}
+
+// Get the next configured response body and error.
+func (g *VariableResponseGetter) Get(_ string) ([]byte, error) {
+	body := g.ResponseBody[g.callCount]
```

Comment:
This construction is error-prone due to the unchecked expectation that the arrays are the same length, and that Get will not be called more times than there are entries. There is an additional test condition I'm not sure you want to test, but that is that all the responses that you prepared got discharged–maybe you want to have a Done(t testing.TB) function that errors if that expectation isn't met.

I also am not keen on the Get semantics being call order dependent across different URLs.

Maybe we can have Responses `map[string][]GetResponse`

with type GetResponse struct {
Occurrences uint
Body []byte
Error error
}

Such that Occurrences is decremented each time it's retrieved, and then if decremented to 0, the array gets the 0th item sliced off.
This makes len(g.Responses[url]) == 0 always a testing internal error that you can return.

Reply:
Thanks - these are great points. PTAL at the updated implementation.

Two more things:

> This makes len(g.Responses[url]) == 0 always a testing internal error that you can return.

When you say return error, do you mean like the `errors.New("404")` you as with the existing Getter? I can also see making this even more strict and `Fail` the test in case there are no configured URLs or no more prepared responses. 

Do you want me to merge the two Getter implementations at this point? I also see value in keeping the simple `Getter` as well, so that tests can use the simpler construction and prevent the verbosity of constructing the responses.

Reply:
I think returning a 404 error is a helpful alternative, to avoid having to code every 404 response yourself. That would allow the two implementations to merge more seamlessly. You can change current &test.Getter{Responses: m} constructions to test.SimpleGetter(m) to construct the more involved type underneath. Occurrences could be set to uint max (^uint(0)) in this case.

Reply:
Works for me! I updated the implementation.